docs/sources/flow/reference/components/otelcol.extension.jaegerremotesampling.md

```diff
@@ -0,0 +1,277 @@
+`auth` | `capsule(otelcol.Handler)` | Handler from an `otelcol.auth` component to use for authenticating requests. | | no
+
+{{< docs/shared lookup="flow/reference/components/otelcol-compression-field.md" source="agent" >}}
+
+The `balancer_name` argument controls what client-side load balancing mechanism
+to use. See the gRPC documentation on [Load balancing][] for more information.
+When unspecified, `pick_first` is used.
+
+An HTTP proxy can be configured through the following environment variables:
+
+* `HTTPS_PROXY`
+* `NO_PROXY`
+
+The `HTTPS_PROXY` environment variable specifies a URL to use for proxying
+requests. Connections to the proxy are established via [the `HTTP CONNECT`
+method][HTTP CONNECT].
+
+The `NO_PROXY` environment variable is an optional list of comma-separated
+hostnames for which the HTTPS proxy should _not_ be used. Each hostname can be
+provided as an IP address (`1.2.3.4`), an IP address in CIDR notation
+(`1.2.3.4/8`), a domain name (`example.com`), or `*`. A domain name matches
+that domain and all subdomains. A domain name with a leading "."
+(`.example.com`) matches subdomains only. `NO_PROXY` is only read when
+`HTTPS_PROXY` is set.
+
+Because `otelcol.exporter.jaeger` uses gRPC, the configured proxy server must be
+able to handle and proxy HTTP/2 traffic.
+
+[Load balancing]: https://github.com/grpc/grpc-go/blob/master/examples/features/load_balancing/README.md#pick_first
+[HTTP CONNECT]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/CONNECT
+
+### tls client block
+
+The `tls` block configures TLS settings used for the connection to the gRPC
+server.
+
+{{< docs/shared lookup="flow/reference/components/otelcol-tls-config-block.md" source="agent" >}}
+
+### keepalive client block
+
+The `keepalive` block configures keepalive settings for gRPC client
+connections.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`ping_wait` | `duration` | How often to ping the server after no activity. | | no
+`ping_response_timeout` | `duration` | Time to wait before closing inactive connections if the server does not respond to a ping. | | no
+`ping_without_stream` | `boolean` | Send pings even if there is no active stream request. | | no
+
+## Component health
+
+`otelcol.extension.jaegerremotesampling` is only reported as unhealthy if given an invalid
+configuration.
+
+## Debug information
+
+`otelcol.extension.jaegerremotesampling` does not expose any component-specific debug information.
+
+## Example
+
+This example configures the Jaeger remote sampling extension to load a local json document and
+serve it over http port 5778. It also disables the default GRPC server hosting the sampling document.
+
+```river
+otelcol.extension.jaegerremotesampling "example" {
+  http {
+    endpoint = "0.0.0.0:5778"
+  }
+  grpc {
+    endpoint = ""
+  }
```

Comment:
I would like to change this so that the presence of the blocks determines which of the servers to start, since setting `endpoint = ""` to disable gRPC would be surprising to me.